### PR TITLE
TIP-858: Enable php coupling detector for user management BC

### PR DESCRIPTION
**Pull request by arnolanglade**
The following drawing present the different bounded contexts we will introduce the next major version. 

dessin sans titre

A bounded context should not be coupled to another one. That's why we use PHP coupling detector to check automatically that. For now, it is now possible to decouple all BC because of our legacy code, we don't to rewrite the PIM/have time. We added some exceptions to the PHP coupling detector configuration but we added comment to explain how fix those coupling issues.

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | -
| Changelog updated                 | -
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -78,7 +78,7 @@
     },
     "require-dev": {
         "akeneo/phpspec-skip-example-extension": "2.0.*",
-        "akeneo/php-coupling-detector": "0.2.0",
+        "akeneo/php-coupling-detector": "dev-master",
```
Comment (damien-carcel) [inline composer.json]: or we could tag a 0.3? :-)

File: src/Akeneo/UserManagement/Component/Model/UserInterface.php [hunk 2]
```diff
@@ -46,7 +45,7 @@ public function getEmail();
     /**
      * Return birthday
      *
-     * @return DateTime
+     * @return \DateTime
```
Comment (anaelChardan) [inline src/Akeneo/UserManagement/Component/Model/UserInterface.php]: Why is it part of this PR?

Comment (arnolanglade) [reply src/Akeneo/UserManagement/Component/Model/UserInterface.php]: I fixed some stuff with more strict rules.

File: composer.json (hunk 1 above)
Comment (arnolanglade) [reply composer.json]: It is not plan for now.

Comment (damien-carcel) [reply composer.json]: OK then.

Comment (jjanvier) [reply composer.json]: I will do it soon :)

**Pull request merged**